test/integration/puppetlabs/services/jruby/jruby_puppet_pool_int_test.clj

```diff
@@ -249,6 +249,8 @@
 (deftest ^:integration test-503-when-jruby-is-first-to-shutdown
   (testing "During a shutdown requests result in 503 http responses"
     (bootstrap/with-puppetserver-running-with-mock-jrubies
+     "MOCK.TODO: discuss this one; it's probably safe because we don't really care
```

Comment:
With this test, I think we're just interested in seeing that after the jruby-puppet-pooled-service is shutdown, any requests to borrow an instance from the pool return a 'service-unavailable' - translated to an HTTP 503 at the HTTP request layer.  All of the logic involved in emptying the pool and dumping a shutdown pill into it so that a borrow gets the pill and throws the 'service-unavailable' in response is still run even if the mock-jruby-pool-manager-service is being used.

So long as the mock requests would return a successful response before the JRuby service is stopped - i.e., anything other than a 503 error - I think the rest of the production code that this test is trying to validate is all still being exercised.

So, yeah, I'm basically okay with this as a mock for the reason you mentioned.  If we decide we want to change `.handleRequest` for the mock JRubyPuppet to throw an exception, though, this test would probably no longer work as written because the pre-shutdown environments requests would start failing.